provider/provider.go

```diff
@@ -94,3 +94,13 @@ func getDefaultPath() string {
 		return "/usr/local/lib/summon"
 	}
 }
+
+//GetAllProviders creates slice of all file names in the default path
+func GetAllProviders() ([]string, error){
+	files, err := ioutil.ReadDir(DefaultPath)
```

Comment:
I think you want to check for error immediately instead of iterating over the files